internal_worker_base.go

```diff
@@ -214,7 +218,8 @@ func (bw *baseWorker) pollTask() {
 	var err error
 	var task interface{}
 	bw.retrier.Throttle()
-	if bw.pollRateLimiter.Consume(1, time.Millisecond*100) {
+	if bw.pollLimiter.Wait(bw.limiterContext) == nil {
```

Comment:
this changes behavior a bit. Previously, it looks like we waited for 100ms and then initiated backoff. Now we wait infinitely. This is fine, except now, the backoff logic is useless. It also doesn't look like backoff belongs here, so you might as well remove it.